portal-impl/test/unit/com/liferay/portal/util/FileImplTest.java

```diff
@@ -33,6 +45,28 @@ public class FileImplTest {
 	public static final LiferayUnitTestRule liferayUnitTestRule =
 		LiferayUnitTestRule.INSTANCE;
 
+	@Before
+	public void setUp() throws Exception {
+		FastDateFormatFactoryUtil fastDateFormatFactoryUtil =
+			new FastDateFormatFactoryUtil();
+
+		fastDateFormatFactoryUtil.setFastDateFormatFactory(
+			new FastDateFormatFactoryImpl());
+
+		_tempFolder = _fileImpl.createTempFolder();
+
+		_evilFileTargetDir = _getTempTestFilePath(
+			"../../../../../../../../../../../../../../../../../../../../.." +
```

Comment:
@jeffwu0724 How did you get this hard coded path here?

Reply:
Hi @tinatian ,

For this ticket, there is a similar unit test in "modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/sidecar/UncompressUtilTest.java" for zipslip.

He created a test zip file "modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/resources/com/liferay/portal/search/elasticsearch7/internal/sidecar/root/test_slip.zip" for this case. If we extract it, it will finally give us "../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../tmp/evil.txt". 

So back in the unit test, if my check for zipslip works, then it will never extra the zip file in this way, and will not find this evil.txt.

Thanks for checking, Tina.

Reply:
@jeffwu0724 Why you extract a test zip stored in another module? You should always make your test independent of other modules, or else if someone remove/rename that zip file, your test is going to fail too.

Reply:
@jeffwu0724 Slack me tomorrow morning, let's have a talk about this.